src/sql/workbench/contrib/resourceViewer/browser/resourceViewerEditor.ts

```diff
@@ -0,0 +1,176 @@
+	private _header: HTMLElement;
+	private _actionBar: Taskbar;
+	private _resourceViewerTableEditor: ResourceViewerTableEditor;
+	private _stateListener: IDisposable;
+
+	private _resourceViewerEditorContextKey: IContextKey<boolean>;
+
+	private _savedTableViewStates = new Map<ResourceViewerInput, ProfilerTableViewState>();
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IWorkbenchThemeService themeService: IWorkbenchThemeService,
+		@IInstantiationService private _instantiationService: IInstantiationService,
+		@IContextKeyService private _contextKeyService: IContextKeyService,
+		@IEditorService editorService: IEditorService,
+		@IStorageService storageService: IStorageService
+	) {
+		super(ResourceViewerEditor.ID, telemetryService, themeService, storageService);
+		this._resourceViewerEditorContextKey = CONTEXT_PROFILER_EDITOR.bindTo(this._contextKeyService);
+
+		if (editorService) {
+			editorService.overrideOpenEditor({
+				open: (editor, options, group) => {
+					if (this.isVisible() && (editor !== this.input || group !== this.group)) {
+						this.saveEditorViewState();
+					}
+					return {};
+				}
+			});
+		}
+	}
+
+	protected createEditor(parent: HTMLElement): void {
+		this._container = document.createElement('div');
+		this._container.className = 'resource-viewer';
+		parent.appendChild(this._container);
+
+		this._createHeader();
+
+		let tableContainer = this.createResourceViewerTable();
+
+		this._container.appendChild(tableContainer);
+	}
+
+	private _createHeader(): void {
+		this._header = document.createElement('div');
+		this._header.className = 'resource-viewer-header';
+		this._container.appendChild(this._header);
+		this._actionBar = new Taskbar(this._header);
+
+		this._actionBar.setContent([
+			// TODO - chgagnon add actions
+		]);
+	}
+
+	private createResourceViewerTable(): HTMLElement {
+		let resourceViewerTableContainer = document.createElement('div');
+		resourceViewerTableContainer.className = 'resource-viewer-table monaco-editor';
+		resourceViewerTableContainer.style.width = '100%';
+		resourceViewerTableContainer.style.height = '100%';
+		resourceViewerTableContainer.style.overflow = 'hidden';
+		resourceViewerTableContainer.style.position = 'relative';
+		let theme = this.themeService.getColorTheme();
+		if (theme.type === DARK) {
+			DOM.addClass(resourceViewerTableContainer, VS_DARK_THEME);
+		} else if (theme.type === HIGH_CONTRAST) {
+			DOM.addClass(resourceViewerTableContainer, VS_HC_THEME);
+		}
+		this.themeService.onDidColorThemeChange(e => {
+			DOM.removeClasses(resourceViewerTableContainer, VS_DARK_THEME, VS_HC_THEME);
+			if (e.type === DARK) {
+				DOM.addClass(resourceViewerTableContainer, VS_DARK_THEME);
+			} else if (e.type === HIGH_CONTRAST) {
+				DOM.addClass(resourceViewerTableContainer, VS_HC_THEME);
+			}
+		});
+		this._resourceViewerTableEditor = this._instantiationService.createInstance(ResourceViewerTableEditor);
```

Comment:
Whats the value here in having an editor inside and editor?

Reply:
Removed the editor stuff and just made it a wrapper over the base table component